src/util/hash.h

```diff
@@ -20,13 +20,10 @@
 #ifndef __CVC4__HASH_H
 #define __CVC4__HASH_H
 
-// in case it's not been declared as a namespace yet
-namespace __gnu_cxx {}
+#include <functional>
+#include <string>
 
-#include <ext/hash_map>
-#include <ext/hash_set>
-
-namespace __gnu_cxx {
+namespace std {
 
 #ifdef CVC4_NEED_HASH_UINT64_T
 // on some versions and architectures of GNU C++, we need a
```

Comment:
Actually, the current version of the pull request fails to compile on macOS because the test in configure.ac fails to detect `std::hash<uint64_t>`, which leads to an error when compiling due to redefining `std::hash<uint64_t>`. So at least the test needs to be updated.